framework/src/org/checkerframework/common/signatureinference/SignatureInferenceScenes.java

```diff
@@ -0,0 +1,726 @@
+package org.checkerframework.common.signatureinference;
```

Comment:
There is a complication if the option -AuseDefaultsForUncheckedCode=source is turned on.
See the manual in the main repository for information on this option (the online manual is not up to date).
In your repository, this option has the old name -AuseSafeDefaultsForUnannotatedSourceCode.

Suppose the following source is being type checked with the Nullness Checker with signature inference and -AuseDefaultsForUncheckedCode=source turned on:

> import org.checkerframework.framework.qual.AnnotatedFor;
> 
> class A {
> 
>   // In unchecked code, return types default to @Nullable, whereas in checked code they default to @NonNull.
>   Object bar() { return new Object(); }
> 
>   Object field = new Object();
> 
>   @AnnotatedFor("nullness")
>   void foo() {
>      field = bar();
>   }
> }


Because bar() is unchecked, the return type of the call to bar() will be @Nullable. The field will then be inferred to have type @Nullable, whereas if the -AuseDefaultsForUncheckedCode=source had not been passed in, it would have been inferred to have type @NonNull.

The code now looks like:

> import org.checkerframework.framework.qual.AnnotatedFor;
> 
> class A {
> 
>   Object bar() { return new Object(); }
> 
>   @Nullable Object field = new Object();
> 
>   @AnnotatedFor("nullness")
>   void foo() {
>      field = bar();
>   }
> }

Finally, the user proceeds to finish annotating their code and moves the @AnnotatedFor to cover the entire class:

> import org.checkerframework.framework.qual.AnnotatedFor;
> 
> @AnnotatedFor("nullness")
> class A {
> 
>   Object bar() { return new Object(); }
> 
>   @Nullable Object field = new Object();
> 
>   void foo() {
>      field = bar();
>   }
> }

Suddenly, the field is @Nullable for no reason, and it is now a "user-written" annotation so that signature inference won't remove it the next time the user runs it.

I see two options:
* Document these subtleties and issue a warning when signature inference is turned on simultaneously with - AuseDefaultsForUncheckedCode=source, or
* Forbid the two options from being used simultaneously